hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/RegionReplicaGroupingCostFunction.java

```diff
@@ -44,11 +46,12 @@ final void prepare(BalancerClusterState cluster) {
 
   protected final long getMaxCost(BalancerClusterState cluster) {
     // max cost is the case where every region replica is hosted together regardless of host
-    int[] primariesOfRegions = new int[cluster.numRegions];
-    System.arraycopy(cluster.regionIndexToPrimaryIndex, 0, primariesOfRegions, 0,
-      cluster.regions.length);
-
-    Arrays.sort(primariesOfRegions);
+    HashMap<Integer, ArrayList<Integer>> primariesOfRegions =
+      new HashMap<Integer, ArrayList<Integer>>(cluster.numRegions);
```

Comment:
We don't do this double allocation on a single line... we give each allocation and assign its own line... easier to read.

Left hand size should be Map rather than HashMap? Ditto for List and ArrayLIst.